Architecture-Proposal/Core/Example/UserFetcher.swift

```diff
@@ -0,0 +1,15 @@
+import Foundation
+import RxSwift
+
+public final class UserFetcher: Fetcher {
+    private let service: APIService
+
+    init(service: APIService) {
+        self.service = service
+    }
+
+    public func fetch(predicate: FetcherPredicate? = nil) -> Observable<User> {
```

Comment:
so you do not need to do `typealias Model == User` in this case? I would assume this doesn't compile because you've not defined the associated type 🤔

Reply:
The protocol signature is `func fetch(predicate: FetcherPredicate?) -> Observable<Model>`
As in this class we are specifying the method signature with the final model, the swift compiler is able to determine the type used in the UserFetcher.